test/puppetlabs/trapperkeeper/services/status/status_service_test.clj

```diff
@@ -176,3 +176,46 @@
                 "message" (str "No status function with version 3 "
                             "found for service foo")}
               (json/parse-string (slurp (:body resp)))))))))
+
+(defn response->status
+  [resp]
+  (:status (json/parse-string (slurp (:body resp)) true)))
+
+(deftest level>=test
+  (testing "use of level>= to implement a status function"
+    (let [my-status (fn [level]
+                      (let [level>= (partial status-core/level>= level)]
+                        {:is-running true
+                         :status (cond-> {:this-is-critical "foo"}
+                                   (level>= :info) (assoc :bar "bar"
+                                                          :baz "baz")
+                                   (level>= :debug) (assoc :x "x"
+                                                           :y "y"
+                                                           :z "y"))}))
```

Comment:
This is the interesting bit of this PR - how would we like the implementation of this status function to look?  This isn't *exactly* what's described in TK-217 (although my reading of that ticket was that it was sort of a rough idea, and that what I came up with here is what @cprice404 was going forward in his description).

I am reasonably happy with how this looks, but would be happy to consider alternative - @cprice404, especially if this isn't exactly what you had in mind.